### PR TITLE
CRM-17361 -- Extend Contact getoptions to include Address fields

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Fixes https://issues.civicrm.org/jira/browse/CRM-17361

### CODE REVIEW

**Comment by colemanw** [comment]
This seems like a good change, at least from the perspective of self-documentation (it should allow the api explorer to better represent these fields).

But since there were no failing tests I assume these fields were not tested? In that case how do we know this change works? Can you please add at least one test to this PR?

File: api/v3/utils.php [hunk 1]
```diff
@@ -1798,6 +1798,9 @@ function _civicrm_api_get_fields($entity, $unique = FALSE, &$params = array()) {
     }
   }
   $fields += _civicrm_api_get_custom_fields($entity, $params);
+  if ($entity == 'Contact') {
+    $fields += _civicrm_api_get_fields('Address');
+  }
```
Comment (colemanw) [inline api/v3/utils.php]: Would be nice to not place hacks for a specific entity in the generic api layer (any more than there already are).
Shouldn't this be in _civicrm_api3_contact_get_spec?

**Comment by colemanw** [comment]
After talking this over with Eileen, we're both wondering what caused these fields to fail in the first place and if there isn't a simpler fix, since this PR is more like adding new features.

I'm not strictly opposed to the new features if they have good test coverage and are cleanly implemented, but still curious as to the original cause of the bug.

**Comment by jitendrapurohit** [comment]
closed in favor of https://github.com/civicrm/civicrm-core/pull/6967.
Moved to master. Will write the test cases and update that PR soon. Thanks!

**Pull request closed without merging**